# DP: Don't build gdc build tools idgen and impcnvgen with profiling flags

--- a/src/gcc/d/Make-lang.in
+++ b/src/gcc/d/Make-lang.in
@@ -322,6 +322,14 @@ d/idgen: d/idgen.dmdgen.o
 d/impcvgen: d/impcnvgen.dmdgen.o
 	+$(LINKER_FOR_BUILD) $(BUILD_LINKER_FLAGS) $(BUILD_LDFLAGS) -o $@ $^
 
+d/idgen.dmdgen.o: d/dmd/idgen.c
+	$(filter-out -fprofile-%,$(DMD_COMPILE)) $(D_INCLUDES) $<
+	$(POSTCOMPILE)
+
+d/impcnvgen.dmdgen.o: $(srcdir)/d/dmd/impcnvgen.c
+	$(filter-out -fprofile-%,$(DMDGEN_COMPILE)) $(D_INCLUDES) $<
+	$(POSTCOMPILE)
+
 # Generated sources.
 d/id.c: d/idgen
 	cd d && ./idgen
